src/plugins/intel_cpu/tests/functional/subgraph_tests/src/matmul_weights_decompression.cpp

```diff
@@ -94,34 +95,64 @@ class MatmulWeightsDecompression : public testing::WithParamInterface<MatmulWeig
     }
 
 protected:
-    std::shared_ptr<ov::Model> initSubgraph(std::vector<ov::PartialShape>& inputShapes,
-                                                const ov::element::Type data_precision,
-                                                const ov::element::Type weights_precision,
-                                                const bool transpose_weights,
-                                                const bool add_subtract,
-                                                const bool reshape_on_decompression) {
-        ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(data_precision, inputShapes[0])};
+    std::shared_ptr<ov::Node> initDecompressionWeights(const ov::Shape& weights_shape,
+                                                       const size_t group_size,
+                                                       const ov::element::Type data_precision,
+                                                       const ov::element::Type weights_precision,
+                                                       const bool transpose_weights,
+                                                       const bool add_subtract,
+                                                       const bool reshape_on_decompression_constant) {
         auto transpose_if_necessary = [&](const ov::Shape& shape) {
-            if (!transpose_weights)
-                return shape;
-            auto transposed_shape = shape;
-            std::swap(*transposed_shape.rbegin(), *(transposed_shape.rbegin() + 1));
-            return transposed_shape;
+            auto result_shape = shape;
+            if (transpose_weights)
+                std::swap(*result_shape.rbegin(), *(result_shape.rbegin() + 1));
+            return result_shape;
         };
 
-        auto weights_shape = transpose_if_necessary(inputShapes[1].to_shape());
-        auto weights = ngraph::builder::makeConstant<uint8_t>(weights_precision, weights_shape, {}, true);
+        // Weights has shape [I, O], where
+        // I - input channels
+        // O - output channels
+        // If group size greater than 1, input channels dimension are split into 2: I -> [N, G], where
```

Comment:
This logic introduces confusion. I expect that in case G = 1, N should be equal IC. Can we use '-1' as default value for G which will trigger test to init N = 1?

Reply:
Default value is introduced, thanks for the suggestion. I also introduced boolean variable `group_decompression` to make it easier to match explanatory comments with the code